api/go.mod

```diff
@@ -1,20 +1,20 @@
 module github.com/ceph/ceph-csi/api
 
 go 1.23.0
+toolchain go1.24.1
```

Comment:
IIRC, @Madhu-1 suggested me to remove this line in some PR. We sure we want to add it with this one? 

_If present it directs the compiler to use the version specified to build instead of the one defined by `go <version>`_

Reply:
It gets added by Dependabot on every PR it creates. Removing it is pretty painful. It doesn't look like there are issues with having it. In case some downstream/product builds don't like it, it can be removed or adjusted there.